environment/src/main/java/com/sequenceiq/environment/environment/flow/start/handler/StartFreeIpaHandler.java

```diff
@@ -39,12 +43,16 @@ public String selector() {
     public void accept(Event<EnvironmentStartDto> environmentStartDtoEvent) {
         EnvironmentDto environmentDto = environmentStartDtoEvent.getData().getEnvironmentDto();
         try {
-            freeIpaService.describe(environmentDto.getResourceCrn()).ifPresent(freeIpa -> {
-                if (freeIpa.getStatus() != null && !freeIpa.getStatus().isStartable()) {
+            freeIpaService.describe(environmentDto.getResourceCrn()).ifPresentOrElse(freeIpa -> {
+                if (freeIpa.getStatus() != null && (freeIpa.getStatus().isAvailable() || freeIpa.getStatus().isStartInProgressPhase())) {
+                    LOGGER.info("FreeIPA is already {}, so it won't started", freeIpa.getStatus());
+                } else if (freeIpa.getStatus() != null && !freeIpa.getStatus().isStartable()) {
```

Comment:
I would move the null check one layer up as it used in two condition paths and in case of start if there is no status for the FreeIPA stack that should be a critical error.

Reply:
Like the previous version, we tried to start the freeipa event if the status was null, I didn't check this. 

I don't know it is a good choice  or not to interrupt the env flow